pkg/tools/log/spanlogger/childspanlogger/childspanlogger.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package childspanlogger contains span logger for specific case of inifinite Find request
+package childspanlogger
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	opentracinglog "github.com/opentracing/opentracing-go/log"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/jaeger"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+)
+
+const (
+	maxStringLength int = 1000
+	dotCount        int = 3
+)
+
+var loggers []*childSpanLogger
+var ticker = time.NewTicker(time.Minute)
+var once sync.Once
+var lock sync.Mutex
+
+// childSpanlogger - provides a way to log via opentracing spans
+type childSpanLogger struct {
+	span      opentracing.Span
+	childSpan opentracing.Span
+	entries   map[interface{}]interface{}
+	lock      sync.RWMutex
+	operation string
+}
```

Comment:
Please let's try to re-use exist span logger to avoid code duplication.

```suggestion
type todoSpanLogger struct {
   *originalSpanLogger
}
```

Reply:
ok, done